### PR TITLE
Make Chain pub(crate)

### PR DESCRIPTION
**Pull request by octol**

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -336,6 +346,38 @@ impl Node {
...
+    /// If our section is the closest one to `name`, returns all names in our section *including
+    /// ours*, otherwise returns `None`.
+    pub fn close_names(&self, name: &XorName) -> Option<Vec<XorName>> {
+        self.chain().and_then(|chain| chain.close_names(name))
+    }
+
+    /// Only if we have a chain (meaning we are elders) we will process this API
+    pub fn min_sec_size_from_chain(&self) -> Option<usize> {
+        self.chain().map(Chain::min_sec_size)
+    }
```
Comment (madadam) [inline src/node.rs]: Again, there already is `Node::min_section_size`. How is this different and why do we need this?

File: src/node.rs [hunk 2]
```diff
@@ -302,15 +302,20 @@ impl Node {
...
     /// Our `XorName`.
     pub fn our_name(&self) -> Option<&XorName> {
         self.chain().map(|chain| chain.our_id().name())
     }
 
-    /// Returns the prefixes of all out neighbours.
+    /// Our ID once we're part of a section
+    pub fn our_id(&self) -> Option<&PublicId> {
+        self.chain().map(Chain::our_id)
+    }
```
[the remark this answers is not in the document]
Comment (octol) [reply src/node.rs]: Is the `id` stored in the chain always guaranteed to be identical to the state one? I mean if it were not true that would point to a problem, but I was being conservative as the above would mean no logic change. But I would also prefer to get rid of it unless you suspect there might be some subtle issues afoot?

File: src/node.rs (hunk 1 above)
Comment (octol) [reply src/node.rs]: Hmm so `Node::min_section_size` appears, unused, should we just remove that one instead?

Comment (octol) [reply src/node.rs]: Added a commit where I remove the unused `Node::min_section_size` and only use the chain backed version

File: src/node.rs [hunk 3]
```diff
@@ -323,6 +328,11 @@ impl Node {
...
     }
 
+    /// Collects prefixes of all sections known by the routing table into a `BTreeSet`.
+    pub fn prefixes(&self) -> BTreeSet<Prefix<XorName>> {
+        self.chain().map(Chain::prefixes).unwrap_or_default()
+    }
+
     /// Returns the members of a section with the given prefix.
     /// Prefix must be either our prefix or of one of our neighbours. Returns empty set otherwise.
     pub fn section_members(&self, prefix: &Prefix<XorName>) -> BTreeSet<XorName> {
```
[the remark this answers is not in the document]
Comment (octol) [reply src/node.rs]: Done!

File: src/node.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (octol) [reply src/node.rs]: Added a commit where I get rid of `our_id` in favour of the already existing `Node::id`

**Pull request merged**